src/classes/HH_NameSpec.cls

```diff
@@ -217,76 +216,102 @@ public with sharing class HH_NameSpec implements HH_INaming {
     * return string
     */
     private static string strNameFromNameSpec(NameSpec ns, list<Contact> listCon) {
-        string strName = '';
-        integer cCon = listCon.size();          
-        integer i = 0;
-        boolean isNoLastnames = (ns.strFullNameSpec.indexOf(chLToken + 'LastName' + chRToken) == -1);
+        Integer size = listCon.size();    
```

Comment:
I will revert name back from "size" to initial "cCon".
Even though "size" might be generic name, it is used as a List method on the platform, so using a variable "size" for listCon.size() seemed ok.